### PR TITLE
Fixes the jsdoc comment format

### LINKED ISSUE #28: After upgrade, it generates invalid jsdoc
I'm not sure what's going on:

```js
/**
 * ParentSearch
 *
 * @name ParentSearch
 * @function
 * @param a {String} Foo
 */
function ParentSearch(a) {
}
```

I expect to get `* @param {String} a Foo`

### PR DESCRIPTION
**Pull request by IonicaBizau**
Fixes #28. Please check if it's fine. Thanks! :smile:

### CODE REVIEW

**Comment by heavenshell** [comment]
Thx for PR.
I'll check it later.

File: autoload/jsdoc.vim [hunk 1]
```diff
@@ -111,7 +111,7 @@ function! jsdoc#insert()
 
   let l:line = getline('.')
   let l:indentCharSpace = ' '
-  let l:indentCharTab = '	'
+  let l:indentCharTab = '       '
```
Comment (heavenshell) [inline autoload/jsdoc.vim]: You should not replace tab to space.
`:set noexpandtab` settings user used `l:indentCharTab`.

Comment (IonicaBizau) [inline autoload/jsdoc.vim]: @heavenshell Whoups, I guess it was replaced by my VIM configs automatically. Fixed. :smile:

**Comment by heavenshell** [comment]
@IonicaBizau 
I've got following result.
```javascript
/**
 * foo
 *
 * @param {} bar
 * @param {} baz
 * @param  {Function}callback The callback function.
 */
function foo(bar, baz, callback) {
}
```

At ` * @param  {Function}callback The callback function.`,
`@param` and `{Function}` has two spaces and no space between `{Function}` and `callback`.

```viml
let g:jsdoc_allow_input_prompt = 1
let g:jsdoc_custom_args_hook = {
  \ 'callback\|cb': {
  \   'type': '{Function}',
  \   'description': 'The callback function.'
  \ },
  \ 'progress': {
  \   'type': '{Function}',
  \   'description': 'The progress function.'
  \ }
  \}
```
Could you fix this?

**Comment by IonicaBizau** [comment]
@heavenshell Reproduced. Could give me a hint where is this handled?

**Comment by heavenshell** [comment]
https://github.com/IonicaBizauKitchen/vim-jsdoc/blob/8cb4556534e4a51dcd6ce3152554a94b249ff4d6/autoload/jsdoc.vim#L87

This line.
Your PR changes to `{type} argname`, so you need add space after `{type}`.
Cheers :smile:

**Comment by IonicaBizau** [comment]
@heavenshell Fixed. :smile: Thanks!

**Comment by IonicaBizau** [comment]
:tada: My first contribution to a vim plugin! :fireworks:

**Comment by heavenshell** [comment]
Merged. thx for your contribution!
Now you can write Vim script :smile:

**Comment by ryanoasis** [comment]
@IonicaBizau Awesome. Welcome to the dark side :smiling_imp: :+1:

**Comment by IonicaBizau** [comment]
@ryanoasis :smile_cat: That was fun! I realize that VIM lang is pretty much intuitive. I have no experience in writing such code, but looking at the other snippets in the file I was kind of easy to fix this. :smile:

**Pull request merged**